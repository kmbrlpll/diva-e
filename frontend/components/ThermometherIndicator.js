import React, {Component } from "react";
import { View, Text, StyleSheet, Button , Image , Dimensions} from "react-native";

const temperaturColorPalette = ['#0054ff' , '#04f06a' , '#f5b700' ,'#e80049','#ba212b'];

function assignThermometerColor (temperature) {
      let t = parseInt(temperature);
      if(t <= 15) {
        return temperaturColorPalette[0]; 
      }else if(t > 15 && t <= 20) {
        return temperaturColorPalette[1];
      }else if(t > 20 && t <= 25) {
        return temperaturColorPalette[2];
      }else if(t > 25 && t <= 30) {
        return temperaturColorPalette[3];
      }else if(t > 30) {
        return temperaturColorPalette[4];
      }
}

function ThermometherIndicator (props) {
	return (
		<View 
        left= {(props.thermometerData.x / props.imageDimensions.x * 100).toString() + "%"} 
        top = {(props.thermometerData.y / props.imageDimensions.y * 100).toString() + "%"}
        key = {props.thermometerData.id}
        style ={props.style}  
        backgroundColor = {assignThermometerColor(props.thermometerData.value)}
        ><Text style={props.textStyle}>{props.thermometerData.value}Â°</Text></View>
	) 
} 

export default ThermometherIndicator;