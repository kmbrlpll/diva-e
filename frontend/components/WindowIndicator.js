import React, {Component } from "react";
import { View, Text , StyleSheet } from "react-native";
import {PulseIndicator} from 'react-native-indicators';

function WindowIndicator (props) {
	const red = '#FF3333';
	return(
		<View 
        style={styles.dot} 
        left= {(props.windowData.x / props.imageDimensions.x * 100).toString() + "%"} 
        top = {(props.windowData.y / props.imageDimensions.y * 100).toString() + "%"}>
	        <PulseIndicator   
	        color = { props.windowData.isOpen ? red : 'transparent'} size = {10}/>
       </View>
	)
}

export default WindowIndicator;

const styles = StyleSheet.create({
	dot: {
	    position:'absolute',
	    marginTop: -5,
	    marginLeft: -5,
	    flex: 1,
 	}
});