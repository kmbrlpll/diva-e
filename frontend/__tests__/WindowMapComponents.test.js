import React from 'react';
import WindowIndicator from '../components/WindowIndicator.js';
import ThermometherIndicator from '../components/ThermometherIndicator.js';

import renderer from 'react-test-renderer';
jest.useFakeTimers(); 

const imageDimensions = { x : 665 , y : 828 };

test('Window indicator renders correctly' , () => {
	const windowMock = {
    id: '1',
    x: 64 , 
    y: 799, 
    isOpen: true,
  };
	const tree  = renderer.create(<WindowIndicator 
		windowData={windowMock} 
		imageDimensions = {imageDimensions} />).toJSON();
	expect(tree).toMatchSnapshot();
});

test('Thermomether indicator renders correctly' , () => {
	const thermometerMock =   {
    id: 1, 
    x: 85,
    y: 734,
    value: '26', 
  };
	const tree  = renderer.create(<ThermometherIndicator 
		thermometerData = {thermometerMock}
		imageDimensions = {imageDimensions}  />).toJSON();
	expect(tree).toMatchSnapshot();
});