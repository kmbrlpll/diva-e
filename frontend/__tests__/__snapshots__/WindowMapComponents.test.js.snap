// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Thermomether indicator renders correctly 1`] = `
<View
  backgroundColor="#e80049"
  left="12.781954887218044%"
  style={
    Object {
      "alignItems": "center",
      "borderRadius": 10,
      "flex": 1,
      "height": 20,
      "justifyContent": "center",
      "marginLeft": -10,
      "marginTop": -10,
      "position": "absolute",
      "width": 20,
    }
  }
  top="88.64734299516908%"
>
  <Text
    style={
      Object {
        "color": "white",
        "fontSize": 8,
      }
    }
  >
    26
    Â°
  </Text>
</View>
`;

exports[`Window indicator renders correctly 1`] = `
<View
  left="9.624060150375941%"
  style={
    Object {
      "flex": 1,
      "marginLeft": -5,
      "marginTop": -5,
      "position": "absolute",
    }
  }
  top="96.49758454106279%"
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "flex": 1,
          "justifyContent": "center",
        },
        undefined,
      ]
    }
  >
    <View
      animating={true}
      animationDuration={1200}
      animationEasing={[Function]}
      color="#FF3333"
      interaction={true}
      renderComponent={[Function]}
      style={
        Object {
          "height": 10,
          "width": 10,
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "bottom": 0,
            "justifyContent": "center",
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          }
        }
      >
        <View
          style={
            Object {
              "backgroundColor": "#FF3333",
              "borderRadius": 5,
              "height": 10,
              "opacity": 0.5,
              "transform": Array [
                Object {
                  "scale": 0.4,
                },
              ],
              "width": 10,
            }
          }
        />
      </View>
      <View
        style={
          Object {
            "alignItems": "center",
            "bottom": 0,
            "justifyContent": "center",
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          }
        }
      >
        <View
          style={
            Object {
              "backgroundColor": "#FF3333",
              "borderRadius": 5,
              "height": 10,
              "opacity": 1,
              "transform": Array [
                Object {
                  "scale": 0.4,
                },
              ],
              "width": 10,
            }
          }
        />
      </View>
    </View>
  </View>
</View>
`;
